File: 1325-remove-leaf-nodes/solution.js
const TreeNode = require('../datastructure/Tree/TreeNode');
const TreeDebugger = require('../datastructure/Tree/TreeDebugger')
/**
 * @param {TreeNode} root
 * @param {number} target
 * @return {TreeNode}
 */
var removeLeafNodes = function(root, target) {
    if(root === null)
        return null
    root.left = removeLeafNodes(root.left, target)
    root.right = removeLeafNodes(root.right, target)
    if(root.val === target && root.left === null && root.right === null)
        return null
    else
        return root
};
if(require.main === module){
    const levelOrder = [1,2,3,2,null,2,4]
    const target = 2
    const root = TreeDebugger.buildLevelOrderTree(levelOrder)
    console.log("Ans = ", removeLeafNodes(root, target))
}

File: 1325-remove-leaf-nodes/Solution.java
import Tree.*;
class Solution1325 {
    public static TreeNode removeLeafNodes(TreeNode root, int target) {
        if (root == null)
            return null;
        root.setLeft(removeLeafNodes(root.getLeft(), target));
        root.setRight(removeLeafNodes(root.getRight(), target));
        if (root.getVal() == target && root.getLeft() == null && root.getRight() == null)
            return null;
        return root;
    }
    static void main(String[] args) {
        Integer[] levelOrder = {1,2,3,2,null,2,4};
        TreeNode root = TreeDebugger.buildLevelOrderTree(levelOrder);
        System.out.println("Ans = " + removeLeafNodes(root, 2));
    }
}